File: Python/Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = format(s)
        if len(s) == 0:
            return True
        for i in range(int(len(s)/2)):
            if not s[i] == s[len(s)-1-i]:
                return False
        return True

def format(s):
    s = s.lower()
    result = ""
    for i in range(len(s)):
        if (0 <= ord(s[i]) - 97 < 26) or (0 <= ord(s[i])-48 <= 9):
            result += s[i]
    print(result)
    return result

File: Java/Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = format(s.toLowerCase());

        int len = s.length();
        for (int i = 0; i < (len/2); i++){
            if (s.charAt(i) != s.charAt(len-1-i)){
                return false;
            }
        }
        return true;
    }
    static String format(String s){
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < s.length(); i++){
            char ch = s.charAt(i);
            if (((int)ch >= 97 && (int)ch <= 122) || (int)ch >= 48 && (int)ch <= 57){
                result.append(ch);
            }
        }
        return result.toString();
    }
}